core/common/src/main/java/alluxio/conf/RemovedKey.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.conf;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nullable;
+
+/**
+ * This class contains old {@link PropertyKey}s which have been removed from use
+ *
+ * This class is used to track keys which were deprecated in previous versions and subsequently
+ * removed in a future version. We still keep them here so that it is possible to provide users
+ * with useful information if they are found to be using an outdated property key.
+ *
+ * Being removed and still used by an application denotes an error.
+ *
+ * @see InstancedConfiguration#validate()
+ * @see PropertyKey#fromString(String)
+ */
+public class RemovedKey {
```

Comment:
nit: since "This class contains old {@link PropertyKey}s which have been removed from use". then shall we call this class `RemovedKeys`?

Reply:
I was following the convention that the class `PropertyKey` is singular, but contains multiple property keys.